packages/strapi-plugin-i18n/services/locales.js

```diff
@@ -1,14 +1,51 @@
 'use strict';
 
-const find = (...args) => strapi.query('locale', 'i18n').find(...args);
+const { setIsDefault } = require('../utils');
 
-const findById = id => strapi.query('locale', 'i18n').findOne({ id });
+const getStore = () =>
+  strapi.store({
+    environment: '',
+    type: 'plugin',
+    name: 'i18n',
+  });
 
-const findByCode = code => strapi.query('locale', 'i18n').findOne({ code });
+const find = async (...args) => {
+  const locales = await strapi.query('locale', 'i18n').find(...args);
 
-const create = locale => strapi.query('locale', 'i18n').create(locale);
+  return setIsDefault(locales);
```

Comment:
I think this formatting should be done in the presentation layer (controller) not the service